### ISSUE TITLE
[BUG] Listing Backups in "Backup" tab fails

### ISSUE THREAD

**Issue by untcha** [longhorn/longhorn#4199]
## Describe the bug

When I click on "Backup" in the Longhorn dashboard. A toast shows up with the following error message:
```
error listing backup volume names: timeout executing: /var/lib/longhorn/engine-binaries/longhornio-longhorn-engine-v1.3.0/longhorn [backup ls --volume-only nfs://<IP-OF-MY-NAS>:/Longhorn], output , stderr, , error <nil>
```
Backups are successfully stored on the NAS in that nfs volume 'Longhorn' and folder 'backupstore'. But do not show up in the Longhorn dashboard.

## To Reproduce

1. Go to 'Settings'
2. Scroll down to 'Backup Target'
3. Insert the backup target 'nfs://<IP-OF-MY-NAS>:/Longhorn'
4. Click on 'Save'
5. Go to 'Backup' tab
7. See error

## Expected behavior

All created backups should be listed

## Log or Support bundle

If applicable, add the Longhorn managers' log or support bundle when the issue happens. 
You can generate a Support Bundle using the link at the footer of the Longhorn UI.

## Environment

 - Longhorn version: 1.3.0
 - Installation method (e.g. Rancher Catalog App/Helm/Kubectl): Helm (flux)
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version: K3s v1.24.2+k3s1
   - Number of management node in the cluster: 3
   - Number of worker node in the cluster: 5
 - Node config
   - OS type and version: Ubuntu 22.04 LTS
   - CPU per node: 4
   - Memory per node: 4 GB
   - Disk type(e.g. SSD/NVMe): Samsung 256 GB 400 MB/s USB 3.1 Flash Drive
   - Network bandwidth between the nodes:
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): Baremetal (Raspberry Pi 4B)
 - Number of Longhorn volumes in the cluster: 1 with 3 replica

**Comment by mantissahz** [comment]
@untcha Could you please check that the nfs server is alive or the network connectivity between work nodes and nfs-server is ok?

**Comment by untcha** [comment]
@mantissahz yes, nfs server is alive and connectivity is available. Longhorn is able to create backups to exact that location but unable to list them in the UI

**Comment by derekbit** [comment]
Probably because "nfsvers" was changed to "mountvers" in Ubuntu 22.04 nfs-common.

**Comment by derekbit** [comment]
Mount options need to be updated on Longhorn's side.
Let us test if this is the culprit and think if there is any workaround.

**Comment by derekbit** [comment]
@untcha Could you try ubuntu 20.04 to see if it can work as expected?

**Comment by untcha** [comment]
@derekbit it did (at least with Ubuntu 21.10)

I stumbled on a similar issue on Ubuntu 22.04 LTS with the nfs-provisioner. I had to add additional mount options:
```
    nfs:
      mountOptions:
      - nfsvers=4.0
```

Only with that the nfs-provisioner was able to work on my cluster.

**Comment by derekbit** [comment]
@untcha Can you show your NFS server kernel version by `uname -r`?

**Comment by derekbit** [comment]
Got it.
I can reproduce this using my own QNAP NAS.
`mount -t nfs -o nfsvers=4.2 10.20.90.100:/Longhorn /root/test`
nfsvers=4.1 or 4.2 hangs the mount operation forever, and server does not return the error code.

Although `nfsvers=4.0` can work correctly, but the `nfsvers` fallback mechanism in Longhorn relies on error code from the server side. No error code returned leads to fallback failure.

I think you can open a ticket with QNAP to raise their awareness of the problem and check if it can provide any solution.

**Comment by derekbit** [comment]
`/var/lib/longhorn-backupstore-mounts/192_168_178_110/Longhorn` already exists. 
You can check if it's already mounted. If it's empty, you can delete it and try again.

**Comment by untcha** [comment]
In the meantime (until fix of QNAP or whatever). Is there a possibility to recover a volume from backup kind of "manually" like the Volume migration pod example?
Starting a pod, mounting the NFS Volume and migrate the backup back into a pvc?

**Comment by derekbit** [comment]
You can copy the backup folder on the QNAP NAS to another machine. Then, export the folder and set the new path to the Longhorn `backup target`.

**Comment by untcha** [comment]
Closing this issue.
Thanks a lot for your effort and help @derekbit 👍🏻
